graphql_compiler/compiler/helpers.py

```diff
@@ -477,3 +520,27 @@ def __ne__(self, other):
     def __hash__(self):
         """Return the object's hash value."""
         return hash(self.base_location) ^ hash(self.fold_path) ^ hash(self.field)
+
+    def _compare(self, other):
+        """Return True if the other object is smaller than self in the total ordering."""
+        if not isinstance(other, FoldScopeLocation):
+            raise AssertionError(u'Expected FoldScopeLocation type for other. Received {}: {}'
+                                 .format(type(other).__name__, other))
+
+        if self.base_location != other.base_location:
+            return self.base_location < other.base_location
+
+        if len(self.fold_path) != len(other.fold_path):
+            return len(self.fold_path) < len(other.fold_path)
+
+        for self_edge_tuple, other_edge_tuple in zip(self.fold_path, other.fold_path):
```

Comment:
Does `if self.fold_path != other.fold_path: return self.fold_path < other.fold_path` not work? If I remember correctly, it's a tuple-of-tuples, which should sort correctly.